ruby/lib/transformer/syskit/exceptions.rb

```diff
@@ -243,10 +261,42 @@ def pretty_print(pp)
     end
 
     # Exception raised when a needed frame is not assigned
-    class MissingFrame < RuntimeError; end
+    class MissingFrame < BaseException; end
 
     # Exception raised when a producer requires itself to function
-    class RecursiveProducer < RuntimeError; end
+    class RecursiveProducer < BaseException; end
 
-end
+    class PortAssociationMismatch < RuntimeError
```

Comment:
Why isn't it a BaseException subclass?

Reply:
Hum, now that I say making this a subclass of BaseException would remove a rescue call specific to PortAssociationMismatch